src/format_csharp.js

```diff
@@ -128,7 +128,7 @@ class CSharpFormatter {
     switch (text) {
       case 'any': return `[object]${optionalSuffix}`;
       case 'unknown': return `[object]${optionalSuffix}`;
-      case 'Array': return `[IEnumerable]${optionalSuffix}`;
+      case 'Array': return direction === 'out' ? `[IReadOnlyList]${optionalSuffix}` : `[IEnumerable]${optionalSuffix}`;
```

Comment:
This mirrors https://github.com/microsoft/playwright/blob/79c4505433c30e63588acc0a5b33d572baa73488/utils/doclint/generateDotnetApi.js#L753.

Reply:
Can we share this logic between the generator and docs?